jsonschema/src/paths.rs

```diff
@@ -41,11 +41,21 @@ impl JSONPointer {
         new.0.extend_from_slice(chunks);
         new
     }
+
     pub(crate) fn as_slice(&self) -> &[PathChunk] {
         &self.0
     }
 }
 
+impl serde::Serialize for JSONPointer {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
```

Comment:
I am wondering whether it would be possible to avoid allocating `String` here and write directly to the underlying receiver & whether it worth the complexity

Reply:
Today I learned about `Serializer::collect`, which does exactly this https://docs.serde.rs/serde/ser/trait.Serializer.html#method.collect_str